Lib/Search/ElasticSearch/Client.php

```diff
@@ -162,9 +163,32 @@ public function removeDocuments(ClassMetadata $class, array $documents)
      */
     public function removeAll(ClassMetadata $class, $query = null)
     {
-        $type = $this->getIndex($class->getIndexForRead())->getType($class->type);
+        $index = $class->getIndexForRead();
         $query = $query ?: new MatchAll();
-        $type->deleteByQuery($query);
+        $this->deleteByScanScroll($class, $query, $index);
+    }
+
+    /**
+     * Perform a scan and scroll through the results, bulk deleting by result ids
+     *
+     * @param ClassMetadata $class
+     * @param AbstractQuery $scanQuery
+     * @param string        $index
+     */
+    private function deleteByScanScroll(ClassMetadata $class, AbstractQuery $scanQuery, $index) {
+        $type = $this->getIndex($index)->getType($class->type);
+        $query = Query::create($scanQuery);
+        $results = $this->scan($query, [$class]);
+
+        foreach($results as $collection) {
+            $resultIds = array();
+            foreach($collection->getResults() as $result) {
+                $resultIds[] = $result->getId();
+            }
+            $this->client->deleteIds($resultIds, $index, $type);
+        }
+        // Refresh index to clear out deleted ID information
+        $this->getIndex($index)->refresh();
```

Comment:
I see, so just make the change to EntityRepository::delete() (and EntityRepositoryCollection::delete(), since that also calls removeAll) to include a $refresh option. I will remove the 'by query' part of those doc blocks.